chore: merge branch 'chore_release-7.0.0' into edge

jerader opened the pull request:
# Overview

Merging `chore_release-7.0.0` into `edge` so engineers working off of `edge` have the labware-adapter split changes among other things.

# Risk assessment

Probably low? Not introducing anything dangerous to edge. Also there were zero merge conflicts somehow.

sfoster1 committed:
build: fix app uploads for release tags and edge

sfoster1 committed:
build: robot builds for all release types (#13306)

* build: robot builds for all release types

* allow other release branch format in robot builds

* truncate changelogs if necessary

* handle limiting the message more readably

sfoster1 committed:
fix(api): update status bar colors for compliance (#13302)

Machinery directive requires specific color patterns for semantic
actions. Update our status bar to be in line with the requirements.

Closes RCORE-819

mjhuff committed:
fix(app): add loading state to protocol routes (#13303)

* feat(app): add loading skeleton to ODD ProtocolDetails route

Adds a shimmer skeleton fallback and prevents Start Setup button selection until the protocol data
loads. Also addresses an edge case concerning components rendering without data after the sever
successfully responds.

* fix(app): prevent protocol setup button resizing when disabled

* fix(app): prevent screen jitter when transitioning from skeleton to page components

Loading skeleton matches the ProtocolSetup components. Refactored skeleton to file and added
testing.

* feat(app): add loading spinner on Start Setup button press

* fix typing errors

* fix lint errors...

* swapped isLoading logic

* fix(app): add skeleton testing for protocol routes

* Review fixes

* final linting...

* Refactored skeleton

SyntaxColoring committed:
test(api,robot-server): Test backwards-compatible validation of metadata and requirements (#13295)

* Add persistence snapshot test, and test for rerunnability.

* Add test for local analysis rejection.

* Add test for POST rejection.

* Rename v6.2.0Large -> v6.2.0_large

koji committed:
fix(app): remove back to RobotDashboard button (#13294)

fix(app): remove back to RobotDashboard button from RunningProtocol and RunSummary

sfoster1 committed:
chore: upload artifacts separate because globs (#13308)

I don't think this action likes having multiple globs as an artifact
target, so do the three separately.

ahiuchingau committed:
add can message to get gripper jaw state (#13296)

SyntaxColoring committed:
feat(api): Add more validation to metadata and requirements dicts (#13258)

Closes RSS-264 and RSS-265.

Add additional validation when parsing Python protocol files:

* Forbid putting `apiLevel` in `metadata` and `requirements` simultaneously. Each protocol should choose one or the other. The historical behavior is that `requirements` overrides `metadata`.
* Forbid unrecognized keys in `requirements`.
* Require `"apiLevel": "2.15"` or newer if  `"robotType"` is `"Flex"` or `"OT-3"`.

smb2268 committed:
fix(app): show robot moving loader when exiting gripper calibration (#13313)

fix RQA-753

smb2268 committed:
fix(app): oDD Run Setup reflect calibration requirements (#13310)

fix RQA-1268

smb2268 committed:
Replace calibration probe image (#13314)

smb2268 committed:
fix(app): memoize pipette name on detach pipette screen (#13316)

fix RQA-1236

ecormany committed:
chore(release): 7.0.0 alpha/beta release notes (#13319)

vegano1 committed:
fix(api): Fix issue where previous cal offsets are used when calibrating the instruments. (#13325)

mjhuff committed:
fix(app): Update receiving new protocol toast (#13326)

* fix(app): update receiving new protocol toast

Adds a 'close' button to the receiving protocol toast and prevents the toast from closing until a
user taps anywhere on the toast. Additionally, all toasts are set 32px from the bottom of the ODD
display.

partially closes RAUT-485

* fix(app): update toast stacking behavior

Partially closes RAUT-485. All ODD toasts now properly play exit animations. Toasts on the ODD have
the option not to stack.

* Fix exitNow prop test

* fix(app): enable hardware acceleration amd optimizations on toast animations

* Old toasts auto fade out when a second toast enters

* Toast refactor

* Delete ternary block

jerader committed:
fix(app): ui changes to moduleInfo and liquid color icon (#13334)

closes RQA-1301 and RQA-1298

SyntaxColoring committed:
fix(robot-server): Advertise a minimum Python Protocol API version of 2.15 from Flex robots (#13329)

Closes RQA-1005.

* Add MIN_SUPPORTED_VERSION_FOR_FLEX.

* Advertise minimum of APIv2.15 for Flex.

koji committed:
feat(app): add home gantry on reset robot settings (#13321)

* feat(app): add home gantry on restart to robotSettings

koji committed:
fix(app): fix robotsettings calibration tab section ui (#13342)

fix(app): The gripper section in calibration tab is not aligned with the design. This PR fix that issue and
modify the space for pipette section.

smb2268 committed:
fix(app): bring users to top of setup when they press Back to Top (#13338)

fix RQA-1305

shlokamin committed:
fix(app-shell): match opentrons usb devices on Windows (#13347)

closes RQA-1317

koji committed:
fix(app): update api support version display format (#13345)

* fix(app): fix the display format of supported protocol api version

jerader committed:
feat(protocol-designer, shared-data, components, app): adapter split support (#13237)

closes RAUT-609 and RAUT-612
closes RAUT-561 and RQA-1076 and RQA-1114

---------

Co-authored-by: Brian Cooper <brian@opentrons.com>

Laura-Danielle committed:
feat(shared-data): Modify shared data to support different liquid 'classes' (#13335)

Laura-Danielle committed:
feat(shared-data): Modify plunger position shape to support different liquid class requirements (#13343)

SyntaxColoring committed:
fix(api): Fix protocol.deck not accepting non-native deck slot spellings (#13320)

Fixes RSS-315.

jerader committed:
fix(app, shared-data): return tip in LPC to default wellLocation (#13353)

closes RAUT-638

---------

Co-authored-by: Brian Cooper <brian@opentrons.com>

jerader committed:
refactor(app): adapter split test coverage and cleanup components (#13351)

closes RAUT-624

jerader committed:
refactor(app, protocol-designer, labware-library): adapter render viewbox adjustments (#13354)

closes RQA-1307 closes RQA-1308

b-cooper committed:
fix(components): if a labware has no wells, labels should default properly (#13358)

Instead of iterating over undefined, default label column to an empty array if no wells on the
targeted labware.

mjhuff committed:
fix(app):  Fix ODD pressed state CSS (#13324)

* fix(app): fix grey bounding boxes

Solves grey background boxes from appearing over certain buttons/navlinks when pressed on the
physical ODD.

* fix(app): add pressed state to ProtocolDetails and ProtocolSetup routes

Some CSS properties cause gray boxes that do not fit the dimensions of the buttons when pressed.
This bug only occurs on the ODD. Removing these CSS properties does not visibly impact styling.

* fix(app): add more missing pressed state

ncdiehl11 committed:
fix(app, labware-library): add new NEST reservoir image (#13301)

* add image for new Nest 1 Reservoir 290ml

* add nest 1 reservoir 290ml image to app labware images

* add nest 1 reservoir 290ml image to labware library labware images

* require nest 1 reservoir 290ml in labware library components

* lint

* re-add optimized images (<2mb)

SyntaxColoring reviewed (approved):
Nice. Looks good to me if CI passes. There were a few runs that were failing and timing out. I've re-run them. Maybe they're just flaky.

smb2268 committed:
fix(app): close maintenance run modal if run gets deleted (#13169)

Fix RQA-1061

mjhuff committed:
feat(app): Desktop: Calibrate banner device detail module card entry point (#13356)

* feat(app): desktop: Add device detail module card "Calibrate Now" banner

Adds a calibrate now banner whenever a specific module requires calibration. Banner display depends
on pipette state: attached & updated firmware, not attached, and needs firmware update. Updates
pipette firmware banner color to error style to reflect update priority. Fixes the right pipette
calibration banner from displaying different text than the left. Includes a slight refactor to make
a few variables more semantic.

Co-authored-by: Nick Diehl <ndiehl@opentrons.com>

* refactor(app): refactor pipette & gripper banners to UpdateBanner component

Co-authored-by: Nick Diehl <ndiehl@opentrons.com>

* app(fix): Module cal possible with a right pipette attached

Co-authored-by: Nick Diehl <ndiehl@opentrons.com>

---------

Co-authored-by: Nick Diehl <ndiehl@opentrons.com>

koji committed:
fix(app): fix RobotSettingsButtons layout and inline notification position (#13362)

* fix(app): fix software update InlineNotification layout

koji committed:
feat(app): update use is robot busy for estop (#13359)

* feat(app): update useIsRobotBusy hook for estop and fix banner rendering issue on OT-2

koji committed:
fix(app): fix NetworkSettings DQA (#13291)

* fix(app): fix NetworkSettings round1 DQA

SyntaxColoring committed:
docs(api,robot-server): Update readmes to avoid overspecifying as "OT-2" (#13357)
